### PR TITLE
Creates the jump off point for creating IR from the TSQL parse tree walker

### PR DESCRIPTION
**Pull request by jimidle**
This is the starting point for new feature issue #275

### CODE REVIEW

**Review by vil1 — COMMENTED** [review]
This looks like duplicate in respect with #267

**Comment by jimidle** [comment]
> This looks like duplicate in respect with #267

Yes - I did search but that did not come up as it seemed like the start of the SELECT statement, and there was no code for TSQL in main at that point. However, this issue will cover more basic elements. We won't get very far into the SELECT statement before we need all the more primitive elements of an expression.

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/expressions.scala [hunk 1]
```diff
@@ -56,7 +56,9 @@ case class Literal(
     float: Option[Float] = None,
     decimal: Option[Decimal] = None,
     double: Option[Double] = None,
+    real: Option[String] = None,
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/expressions.scala:59]: Changing this is not allowed - it's generated from protobuf. Use double

File: core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala [hunk 2]
```diff
@@ -1,20 +1,53 @@
...
   }
 
   override def visitFull_column_name(ctx: Full_column_nameContext): Expression = {
     val columnName = ctx.id_.getText
     Column(columnName)
   }
+
+  override def visitPrimitive_constant(ctx: Primitive_constantContext): Expression = {
+
+    ctx match {
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala:32]: From our scala style:

For method whose entire body is a pattern match expression, put the match on the same line as the method declaration if possible to reduce one level of indentation.

Comment (nfx) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala:32]: Please read this (whole) document again: https://github.com/databricks/scala-style-guide?tab=readme-ov-file#pattern-matching

File: core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala [hunk 3]
```diff
@@ -1,20 +1,53 @@
...
+      case c if c.STRING() != null => c.STRING().accept(this)
+      case c if c.INT() != null => c.INT().accept(this)
+      case c if c.FLOAT() != null => c.FLOAT() accept (this)
+      case c if c.HEX() != null => c.HEX() accept (this)
+      case c if c.REAL() != null => c.REAL() accept (this)
+    }
+  }
+
+  override def visitTerminal(node: TerminalNode): Expression = {
+    val sym = node.getSymbol
```
Comment (nfx) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala:43]: Do we need this constant?..

**Review by nfx — CHANGES_REQUESTED** [review]

**Review by nfx — APPROVED** [review]
Lgtm

**Pull request merged**